main/zoe/guide/contracts/constantProductAMM.md

```diff
@@ -0,0 +1,198 @@
+and `Liquidity` in the `want` section. Removing liquidity reverses the keywords:
+`Liquidity` in the `give` section, and `Central` and `Secondary` in the want
+section. If the proposal specifies amounts directly taken from a recent quote, and
+any trading has intervened, the trade is unlikely to be accepted. You can either
+specify limits on how far the price may have moved, or specify limits of zero and
+trust the contract to trade fairly.
+
+Transactions that don't require an invitation include `addPool()` and the queries
+(`getInputPrice()`, `getOutputPrice()`, `getPoolAllocation()`,
+`getLiquidityIssuer()`, and `getLiquiditySupply()`).
+
+## The ConstantProduct API
+
+These examples use RUN as the central token. BLD and ATM are secondary currencies.
+
+### Trading with the ConstantProduct AMM
+
+Once trading pools have been set up, a new trader can interact with the market by
+asking for the current price, making an invitation, and making an offer. If Sara has
+ATM and needs 275 BLD for a deal she has negotiated, she can use
+getOutputPrice() to get a quote. (An empty amount indicates no limit on the
+amountOut of the result.)
+
+```js
+const quote = E(publicFacet).getOutputPrice(
+  AmountMath.make(BLDBrand, 275n),
+  AmountMath.makeEmpty(ATMBrand),
+);
+  ```
+  
+Let's assume the quote says she needs to provide 216 ATM. Sara believes the
+price is somewhat volatile, and she doesn't want to make repeated calls, so she pads
+her offer. If the appropriate pools don't exist, she'll get an error (`brands were
+not recognized`). If someone sells a lot of ATM into the pool just ahead of
+her, the price will increase, and she'll have to decide whether to deposit more
+ATM or wait for the price to stabilize. If someone buys a lot of ATM just
+ahead of her order, she'll get the 275 BLD for less and will get some ATM
+back.
+
+```js
+const saraProposal = harden({
+  want: { Out: AmountMath.make(BLDBrand, 275n) },
+  give: { In: AmountMath.make(atmBrand, 220n) },
+});
+
+const swapInvitation = await E(publicFacet).makeSwapOutInvitation();
+const atmPayment =
+  harden({ In: saraAtmPurse.withdraw(AmountMath.make(atmBrand, 220n)) });
+
+const saraSeat = await E(zoe).offer(swapInvitation, saraProposal, atmPayment);
+const saraResult = await saraSeat.getOfferResult();
+```
+
+If the result is `Swap successfully completed.`, she got the BLD for 220 ATM
+or less (she'll want to deposit any refund). Otherwise the market price moved against
+her, and she'll have to check the price again and make another offer.
+
+```js
+const BLDProceeds = await E(saraSeat).getPayout('In');
+const atmRefund = await E(saraSeat).getPayout('Out');
+
+const BLDProceedsAmount = E(saraBLDPurse).deposit(BLDProceeds);
+E(saraAtmPurse).deposit(atmRefund);
+```
+
+###  Creating a new Pool
+
+When the contract is first instantiated, there won't be any pools ready for
+trading. `addPool()` adds a new currency, which can then be funded. When a pool is
```

Comment:
Document that the issuer passed to `addPool` must be fungible? (use `NAT` MathKind) Or do we suppose the example suffices?

Also, note or refer to risks around trusting issuers?